Create Meeting 065.md (#182)
ethereum-cat-herders/EIPIP: All EIPIP Meetings/Meeting 065.md

# EIPIP Meeting 65 Notes
### Meeting Date/Time: Wednesday, Sep 21, 2022, at 14:00 UTC
### Meeting Duration: 60 minutes
### [GitHub Agenda Page](https://github.com/ethereum-cat-herders/EIPIP/issues/176)
### [Audio/Video of the meeting](https://youtu.be/0miajHm8JXs)
### Moderator: Pooja Ranjan
### Notes: Alen(Santhosh)

## DECISION ITEMS

**DECISION 65.1**: Victor to create an PR to further discuss on the copyright waiver section instead of mandate the CLA.

----

## Intro [0.00](https://youtu.be/0miajHm8JXs?t=1)
**Pooja Ranjan**

> * Welcome to EIPsIP Meeting 65. I have shared agenda in chat and, we have a few open pull requests and some issues from EIPs Github repository here, for discussion and probably make some decisions so we can move forward with the pull requests at least. And then we get into, the progress on EIPs bot with some EIPs insight and, maybe action items or decision made from the earlier meeting. So let's get started. 

Reviewer: ```suggestion
* Welcome to EIPIP Meeting 65. I have shared agenda in chat and, we have a few open pull requests and some issues from EIPs Github repository here, for discussion and probably make some decisions so we can move forward with the pull requests at least. And then we get into, the progress on EIPs bot with some EIPs insight and, maybe action items or decision made from the earlier meeting. So let's get started. 
```

## CI: Add Markdown Linter ethereum/EIPss#5367 [0.42](https://youtu.be/0miajHm8JXs?t=42)
* The first one listed here is a CI: Add Markdown Linter. I suppose it was added by Sam Wilson. Sam, if you would like to maybe talk about the proposal. 

**Sam**
* Sure. The idea is to add a bot that, corrects common markdown issues and tries to keep people to the lowest common denominator of markdown so that it's compatible with different renderers. there's been a lot of discussion on that pull request already. I think we've settled on a relatively decent set of links to start, and we just wanted to get some, consensus before merging that PR. 

**Pooja Ranjan**
* Sounds good. I don't know if Greg, you have any thoughts to add here? I know you earlier mentioned that it is related to bot, but yeah.

> **Golvin**

Reviewer: ```suggestion
**Greg Colvin**
```

* No, on late breaking agenda, I noticed Sam is here, so it could be an opportunity for he and I to discuss, the reference client and how that's going and how, how we should be relating it to the, the EIPs work, since we're making progress on that as we we clear through these other smaller issues, that would be maybe a good time for us to talk. 

**Pooja Ranjan**

> * Yeah. Okay. I'll, I'll put a note on the agenda for that. 

Reviewer: ```suggestion
* Yeah. Okay. I'll put a note on the agenda for that. 
```

> **Golvin**

Reviewer: ```suggestion
**GColvin**
```

* Okay, good. 

**Pooja Ranjan**

> * Cool. I, I don't see any other EIPs editors, I can't see, Sam, sorry, Lightclient or, or anyone else. Panda pip mention that he would be a little late for the meeting today, however, this Pull request is by him, so I'm assuming that he is definitely on board. So, maybe we can just let people know one more time in discord channel and move ahead with the decision sounds fair?

Reviewer: ```suggestion
* Cool. I don't see any other EIPs editors, I can't see, Sam, sorry, Lightclient or, or anyone else. Pandapip mention that he would be a little late for the meeting today, however, this Pull request is by him, so I'm assuming that he is definitely on board. So, maybe we can just let people know one more time in discord channel and move ahead with the decision sounds fair?
```

## Website: Add MathJax ethereum/EIPss#5596 [3.03](https://youtu.be/0miajHm8JXs?t=185)
**Sam**

> * Yeah, I, I guess, we can go to the next one. The MathJax one.. Yeah, I think that's a, a less, suggestion. It's just adding a JavaScript library that can render late tech. and I think, Greg, you mentioned you like that, so maybe that's a good thing to to add, 

Reviewer: ```suggestion
* Yeah, I guess, we can go to the next one. The MathJax one.. Yeah, I think that's a, a less, suggestion. It's just adding a JavaScript library that can render late tech. and I think, Greg, you mentioned you like that, so maybe that's a good thing to to add, 
```

**Pooja Ranjan**
* So the idea here is like we are trying to, make all editors aware of any changes which are proposed and also being merged, by active editors on board  however, it is good to always collect, kind of rough consensus from all the present editors. So yeah, if you have any thoughts on that. 

**Golvin**
* Yep. That's all. 

**Sam**

> * So you're, are you okay or not okay with, adding math checks? 

Reviewer: ```suggestion
* So you're, are you okay or not okay with, adding mathJax? 
```

> **Golvin**

Reviewer: ```suggestion
**GColvin**
```

* I'm just asking whether it renders properly at the end of the line. 

**Sam**
* Oh, I'm not sure. I didn't hear you ask anything. Yeah, I've never used it. I think it's probably okay. I think it's the most common JavaScript, latex renderer, but, we can, we can do some tests if You'd like. 

> **Golvin**

Reviewer: ```suggestion
**GColvin**
```

* Yeah, I mean, that's the big question. The last time I tried to use it, I couldn't get it to work, but that doesn't mean it doesn't work. It just means I couldn't get it to work.

**Sam**
* Yeah. No, I, I know how that it goes. I guess if there's anything in particular you'd like to see, like a test of, put it in a comment on that PR and we can try it out. 

**Golvin**
* Okay. I mean, EIPs are full of, of math that winds up getting, you know, sort of butchered with, with, you know, asking math in line. that would looks much better if it was, you know, properly type set, and snatches from the yellow paper and things like that. 

**Sam**

> * Yeah. Okay. Well, I'll, I'll try to get it to format some stuff from, yellow paper. Yeah. 

Reviewer: ```suggestion
* Yeah. Okay. Well, I'll try to get it to format some stuff from, yellow paper. Yeah. 
```

**Golvin**

> * Yeah. Maybe one of the other formats works better. I really don't know. I, I don't know where in the pipeline that final, that final, copy gets rendered, by what system, is the name of the renderer we use. Okay. So somewhere there's documentation for the, the dialect of markup that it accepts. 

Reviewer: ```suggestion
* Yeah. Maybe one of the other formats works better. I really don't know. I don't know where in the pipeline that final, copy gets rendered, by what system, is the name of the renderer we use. Okay. So somewhere there's documentation for the, the dialect of markup that it accepts. 
```

**Sam**
* Yep. Somewhere I, I've never looked Really, 

**Golvin**
* Really good system back in the seventies. It, it had its quirks too. It it definitely did. 

**Sam**
* Right. But, you know, they all do. We knew what they were. Yeah, that's true. So I don't, we have enough editors on today to talk about adding a new editor. I don't feel like two of us is sufficient, and I think, 

**Golvin**
* I think he's done enough good work that there's, I can't imagine any objection, but Okay. 

**Sam**

> * Yeah, I'm, I'm good with that then. 

Reviewer: ```suggestion
* Yeah, I'm, good with that then. 
```

**Sam**
* I think we can probably make that comment on a PR, but really it's, that's up to the author. I think that's like a, more of a content thing than a like formatting thing. But yeah, we can definitely suggest it. 

**Victor**
* Yeah. I feel it's editorial because it's, like if you want people to adopt your EIPs, you have to be specific in, in a type, so that the client implementing your EIPs is consistent across the board. 

**Pooja Ranjan**
* All right. So, I think, Sam's suggestion may work here. We can probably leave a comment on the pull request and that author decide on that. but going back to, the discussion of executions spdec, if I remember correctly, a few meetings earlier, we were on a page where we were hoping to have a meta E for the changes those are proposed. I am wondering, if we are there yet, or, we still see some discussion needed to be reaching that proposaling status. 

**Sam**
* We have the same status of the hack MD that Tim put up. we haven't really progressed beyond that yet because like I said, the merge and and all core dev  is paused for the time being.

**Pooja Ranjan**

> * Yeah, no, I'm gonna share the link here, or you already did? No. Okay. I have shared the link to the Hack md, that, Tim has, shared earlier for people to refer to. I believe the changes has to be done on both the el, ml sites, so we should look forward to the meta proposal that is coming up for the expected changes with executable spec. Thank you for sharing all updates. 

Reviewer: ```suggestion
* Yeah, no, I'm gonna share the link here, or you already did? No. Okay. I have shared the link to the [Hackmd](https://notes.ethereum.org/@timbeiko/executable-eips), that, Tim has, shared earlier for people to refer to. I believe the changes has to be done on both the EL and CL sides, so we should look forward to the meta proposal that is coming up for the expected changes with executable spec. Thank you for sharing all updates. 
```

## Issues section [14.38](https://youtu.be/0miajHm8JXs?t=878)
* Sam. Moving on to the issue sections. I have listed a few issues which I collected from the EIPss Github repository. A few of them are created by Panda. I know he is not on the call, but I just wanted to share with all the, present participants if they have any thoughts that they would like to share or else they are always welcome to leave a comment there. 

> ## New contract interface and EIPss should be required to include a privacy considerations section ethereum/EIPss#5682 [15.16](https://youtu.be/0miajHm8JXs?t=916)

Reviewer: ```suggestion
## New contract interface and EIPs should be required to include a privacy considerations section ethereum/EIPss#5682 [15.16](https://youtu.be/0miajHm8JXs?t=916)
```

> * The first one listed here is new contract interface and EIPs should be required to include a privacy concentration section. This is by, a new user. Yeah, anyone has any thought on this? 

Reviewer: ```suggestion
* The first one listed here is new contract interface and EIPs should be required to include a privacy consideration section. This is by, a new user. Yeah, anyone has any thought on this? 
```

**Sam**
* I think we're a blockchain and we're not a shielded blockchain, so I don't think, we really need to write privacy sections, but that's just my opinion. 

**Golvin**
* I see there are, I am not a lawyer problem here. 

**Pooja Ranjan**

> * That's a good point.. All right. yeah, I mean, like obviously if, if there are some changes that people are interested in proposing any changes, the next step would be to propose it in form pull request. Creating an issue definitely gives a good start so people can start adding thoughts over there. Yep. And I think the next one, which is, EIPss should preemptively add a CLA is also from the same author, I believe. 

Reviewer: ```suggestion
* That's a good point. All right. yeah, I mean, like obviously if, if there are some changes that people are interested in proposing any changes, the next step would be to propose it in form pull request. Creating an issue definitely gives a good start so people can start adding thoughts over there. Yep. And I think the next one, which is, EIPs should preemptively add a CLA is also from the same author, I believe. 
```

* So even if you, so a client has to make their, the implementing client have to make their own judgment and research. If they're serious about this, I, I'm in favor of the practice that we can recommend, that's EIPs proposal to say that they release the patents. So my take on the EIPs is that it's an improvement proposal. Therefore, if you want to be adopted as an author, then the easiest for you to like, the more, the easier you make it to adopt it, the better for you as an author and wearing a hat of, of like editorial or editorial contributor, I would say making authors from different background, the easiest to contribute the better for the entire EIPs repository. So that, I would say using that argument, I was trying to advocate that we can recommend, but not make it a, a mandate. 

**Pooja Ranjan**
* Okay. And I also have shared the link to, a document created by Panda and favor of, this proposal. So yeah, people may go through, probably leave their more comments there. 

**Golvin**
* I agree with that. I definitely agree. 

**Sam**

> * Sorry, sorry. The recommendation is to create an optional CLA or 

Reviewer: ```suggestion
* Sorry. The recommendation is to create an optional CLA or 
```

**Victor**
* Should the, the CLA issue be a good place to continue the discussion? 

**Sam**
* Yeah, Yeah. I'm just saying like a PR would be good to actually put the, like decide the actual wording. 

**Victor**
* Okay. Okay. That sounds good. Yeah, I can do that. 

**Sam**
* Great. Thanks. 

## Errata'd EIPss ethereum/EIPss#5652 [21.38](https://youtu.be/0miajHm8JXs?t=1298)
**Pooja Ranjan**
* Thank you. Moving on. The next one is EIPs. It is proposed by Panda. Looks like a new one, not really new seven days ago, but there is a proposed change, to EIPs number if anyone would like to take a look or have any opinion on thoughts on that. The number is 5652. 
* Okay. we can, if anyone has any thoughts, they would like to share right away, it's fine. We can, we can discuss that. Otherwise, have shared the link, probably we can add comment over there. 

## Dynamic Required Approvals ethereum/EIPss#5651 [22.30](https://youtu.be/0miajHm8JXs?t=1350)
* The next one is a dynamic required approval. So there is some proposal of, okay, how many approvals should be required. I'm gonna share the link here. that is 5651 issue number 5651 again, by Panda. Yeah, I would encourage you. 

**Sam**

> * So this is, this is the more complicated, approval system that Micah had proposed. I don't think I have any problems with doing this. I just think it's complicated. 

Reviewer: ```suggestion
* So this is the more complicated, approval system that Micah had proposed. I don't think I have any problems with doing this. I just think it's complicated. 
```

## What license to use for EIPs assets [29.57](https://youtu.be/0miajHm8JXs?t=1794)
**Pooja Ranjan**
* That sounds fair. And I believe the next topic that is listed as here for formal discussion is about license and yeah, what license do use for EIPs assets? We discussed this in the past, but I believe there were no, conclusion or decision on that, and we wanted to have a little bit more of discussion here. So, yeah. some, if you would like to give a little bit background on this topic, just jog the memory of everyone here. 

**Sam**
* Some editors want to force all assets to be cc0. And I, I don't want to do that. I want anything that isn't, copy left to be allowed for assets. 

**Pooja Ranjan**
* Greg, do you have any thoughts on this? 

**Golvin**
* I'm not a lawyer. 

**Pooja Ranjan**

> * Again,. All right. I mean, on a high level, I personally think that when we are talking Ethereum and we are talking about having something as a standard and giving place Ethereum repository, we should try to make, things, as much as open source, free, no patent, no, like trademark things, so people can use it freely, Ethereum is an open source blockchain. We are like trying to, I mean, anyone is welcome to come and use it, so why to add things which may or may not have, permissions to be used or reused, actually. But that's my thought on that. I would be happy to hear other, editors thought on that. 

Reviewer: ```suggestion
* All right. I mean, on a high level, I personally think that when we are talking Ethereum and we are talking about having something as a standard and giving a place in Ethereum repository, we should try to make things as much as open source, free, no patent, no, like trademark things, so people can use it freely, Ethereum is an open source blockchain. We are like trying to, I mean, anyone is welcome to come and use it, so why to add things which may or may not have, permissions to be used or reused, actually. But that's my thought on that. I would be happy to hear other, editors thought on that. 
```

**Sam**

> * Yeah, that, that, that's my thought as well. Like if somebody wants to come in with a, and and mandate that you have to use a certain algorithm in an EIPs and then, you know, a year later they start suing everybody, because they patented that algorithm, I think that's completely ridiculous and we should try to avoid that situation. 

Reviewer: ```suggestion
* Yeah, that's my thought as well. Like if somebody wants to come in with a mandate that you have to use a certain algorithm in an EIPs and then, you know, a year later they start suing everybody, because they patented that algorithm, I think that's completely ridiculous and we should try to avoid that situation. 
```

**Pooja Ranjan**
* Right. And, now we are joined by Pandapip, Panda. We were discussing a couple of proposals, by you, issues that you have listed, and one of the latest one, which is currently in a discussion is about the, patent and also the use of license for EIPs assets. I wonder if you have any thoughts to share on that? 

**Pandapip**
* Sorry, didn't catch that second part. 

**Pooja Ranjan**

> * So it's about the, license usage for EIPss

Reviewer: ```suggestion
* So it's about the, license usage for EIPs
```

**Pandapip**
* I'm right next to a road right now when I'm outside and on really, really, really bad internet and a bad computer. 

**Pooja Ranjan**

> * Okay, Not a problem. if you can hear me now, we were discussing about, if we should be allowing patenting, staff or maybe assets, which may have some trademark already or should it be only, which is free for people to reuse license issues. If you have any thoughts on that, probably you can share. 

Reviewer: ```suggestion
* Okay, Not a problem. if you can hear me now, we were discussing about, if we should be allowing patenting, stuff or maybe assets, which may have some trademark already or should it be only, which is free for people to reuse license issues. If you have any thoughts on that, probably you can share. 
```

**Pandapip**

> * Well, I think it's pretty obvious that, well, I mean, for one thing, all the, all of the, assets should be, sorry, all the EIPs should be cc0, just so that anyone can just use an EIPs  without having to worry about randomly being sued, like in any case. and I feel like patent and trademark, waivers are just like kind of an extension of that. You should be able to use an EIPs without having to worry about randomly being sued for using it. 

Reviewer: ```suggestion
* Well, I think it's pretty obvious that, well, I mean, for one thing, all of the, assets should be, sorry, all the EIPs should be cc0, just so that anyone can just use an EIPs  without having to worry about randomly being sued, like in any case. and I feel like patent and trademark, waivers are just like kind of an extension of that. You should be able to use an EIPs without having to worry about randomly being sued for using it. 
```

**Golvin**
* We, yeah, we specifically, we specifically say that, that we have nothing to do with them. so I'm generally in favor of just leaving it as it is. I hate putting more requirements on authors and we haven't had any problem yet. 

**Victor**
* I feel similar with Greg. I, I can see the value of, requiring patent to be granted or be disclosed. I feel it's a little bit too, too much to ask. And also, you can't prevent someone who propose an EIPs and influence someone else's patent, if they're not aware of. So I think it's general. 

**Pandapip**
* That's what I'm, what I'm suggesting is that basically the bat wave I've added is that the authors can't sue people for, for their own patents. Other people can other, anytime that it's a patent not owned by the author, it's like, say like some, some random third party, they use some random third party, image image, thing that's, that is, patented. The authors had no idea, patent waiver wouldn't apply there, since it's specifically the authors is authors is, right set are waved. Not those of the third party.

**Victor**

> * I'm in, I'm in favor of, have it optional and make it a convenient way for people to wave the, the patent. because as author, if, if it's a sincere author, the easiest it, it is for the adopters, the better for the EIPs. So any sincere author would try to make it as easy as possible. But I, coming from an industry that cares a lot about patent and EIPss, for like, for, for people like me, I'm, I'm not even sure if I were able to contribute to EIPs anymore if there is a mandate for patent granting because of, other type, agreements that I have with, for example, like my employer. So that, that's the background. So I, I when, when Greg mentioned that, in Oracle employees were not even permitted to make, patent search, I, I totally resonate because that's some other trainings I've seen as well. So I, I think we're entering a space of legal practice and, none of us, I, I assuming is a lawyer here on this call. I, I can see the value and a good heart behind it. but I, I think we, we need a little bit more, We might need a little bit more expertise, in the, in a call for, for making it such a strong decision. I, I think optional is definitely valuable. Like making it optional I think is definitely valuable to be, 

Reviewer: ```suggestion
* I'm in favor of, have it optional and make it a convenient way for people to wave the, the patent. because as author, if, if it's a sincere author, the easiest it, it is for the adopters, the better for the EIPs. So any sincere author would try to make it as easy as possible. But I, coming from an industry that cares a lot about patent and EIPs, for like, for, for people like me, I'm, I'm not even sure if I were able to contribute to EIPs anymore if there is a mandate for patent granting because of, other type, agreements that I have with, for example, like my employer. So that, that's the background. So, I when, when Greg mentioned that, in Oracle employees were not even permitted to make, patent search, I, I totally resonate because that's some other trainings I've seen as well. So I, I think we're entering a space of legal practice and, none of us, I, I assuming is a lawyer here on this call. I, I can see the value and a good heart behind it. but I, I think we, we need a little bit more, We might need a little bit more expertise, in the, in a call for, for making it such a strong decision. I, I think optional is definitely valuable. Like making it optional I think is definitely valuable to be, 
```
